internal/gatewayapi/backendtlspolicy.go

```diff
@@ -115,28 +118,31 @@ func (t *Translator) applyEnvoyProxyBackendTLSSetting(policy *gwapiv1a3.BackendT
 		ancestorRefs := []gwapiv1a2.ParentReference{
 			parent,
 		}
+		var err error
 		if ns != ep.Namespace {
+			err = fmt.Errorf("ClientCertificteRef Secret is not located in the same namespace as Envoyproxy. Secret namespace: %s does not match Envoyproxy namespace: %s", ns, ep.Namespace)
 			status.SetTranslationErrorForPolicyAncestors(&policy.Status,
 				ancestorRefs,
 				t.GatewayControllerName,
 				policy.Generation,
-				status.Error2ConditionMsg(fmt.Errorf("client authentication TLS secret is not located in the same namespace as Envoyproxy. Secret namespace: %s does not match Envoyproxy namespace: %s", ns, ep.Namespace)))
-			return tlsConfig
+				status.Error2ConditionMsg(err))
```

Comment:
Non-blocking: we probably should set the status of the EnvoyProxy instead of BackendTLSPolicy as the `ClientCertificateRef` is not part of the `BackendTLSPolicy`.  WDYT @arkodg ?

Reply:
we haven't added a status field yet in `EnvoyProxy` and continue to use the linked resource as the place to update status